prow/tide/tide.go

```diff
@@ -408,6 +408,7 @@ func (c *Controller) Sync() error {
 	c.sc.requiredContexts = requiredContextsMap(filteredPools)
 	select {
 	case c.sc.newPoolPending <- true:
+		c.sc.dontUpdateStatus.reset()
```

Comment:
What kind of guarantees do we have about the overall memory footprint of the set given how often we expect to call this?

Reply:
This set only contains PRs that we merged and is reset after status controller sync finished + tide sync finished it querying. The mem usage of this is absolutely negligible.